dotCMS/src/main/java/com/dotcms/rest/api/v1/asset/WebAssetHelper.java

```diff
@@ -234,6 +250,94 @@ FolderView toAssetsFolder(final Folder folder) {
                 .build();
     }
 
+    public WebAssetView saveOrUpdateAsset(FileUploadData form,  final User user)
+            throws DotDataException, DotSecurityException, IOException {
+
+        final FormDataContentDisposition contentDisposition = form.getContentDisposition();
+        final String fileName = contentDisposition.getFileName();
+        final InputStream fileInputStream = form.getFileInputStream();
+        final FileUploadDetail detail = form.getDetail();
+        final String assetPath = detail.getAssetPath();
+        final ResolvedAssetAndPath assetAndPath = AssetPathResolver.newInstance()
+                .resolve(assetPath, user, true);
+
+        final Host host = assetAndPath.resolvedHost();
+        final Folder folder = assetAndPath.resolvedFolder();
+        final String assetName =  fileName != null ? fileName : assetAndPath.asset();
+
+
```

Comment:
Not sure if the update a file should follows the same rules as temp api does

Reply:
done